server/interpreter.js

```diff
@@ -4931,20 +4971,21 @@ stepFuncs_['Literal'] = function (stack, state, node) {
  * @return {!Interpreter.State|undefined}
  */
 stepFuncs_['LogicalExpression'] = function (stack, state, node) {
-  if (node['operator'] !== '&&' && node['operator'] !== '||') {
-    throw SyntaxError('Unknown logical operator: ' + node['operator']);
-  }
-  if (!state.doneLeft_) {
-    state.doneLeft_ = true;
+  if (state.step_ === 0) {  // Eval left.
+    state.step_ = 1;
     return new Interpreter.State(node['left'], state.scope);
   }
-  if (!state.doneRight_ &&
-      ((node['operator'] === '&&' && state.value) ||
-      (node['operator'] === '||' && !state.value))) {
-    // No short-circuit this time.
-    state.doneRight_ = true;
-    return new Interpreter.State(node['right'], state.scope);
+  if (state.step_ == 1) { // Check for short-circuit; eval right.
```

Comment:
===
Also search for '} //' and add a space.

Reply:
All done.